### PR TITLE
Import NTT Repo to BC GOV

### PR DESCRIPTION
**Pull request by ekwanNTT**
Init SCSS project

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+FROM openjdk:11
+
+COPY ./target/scss-application.jar scss-application.jar
+
+ENTRYPOINT ["java","-jar","/scss-application.jar"]
```
Comment (HypnoMama) [inline Dockerfile:5]: New line at end of file is required

File: src/main/java/ca/bc/gov/open/Scss/Configuration/SecurityConfig.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@EnableWebSecurity
+@Configuration
+public class SecurityConfig extends WebSecurityConfigurerAdapter {
+
+    @Value("${security.basic-auth.username:}")
```
Comment (HypnoMama) [inline src/main/java/ca/bc/gov/open/Scss/Configuration/SecurityConfig.java:18]: Please delete colon after username
```suggestion
    @Value("${security.basic-auth.username}")
```

File: src/main/java/ca/bc/gov/open/Scss/Configuration/SoapConfig.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+import org.springframework.ws.wsdl.wsdl11.DefaultWsdl11Definition;
+import org.springframework.xml.xsd.SimpleXsdSchema;
+import org.springframework.xml.xsd.XsdSchema;
+
+@EnableWs
+@Configuration
+public class SoapConfig extends WsConfigurerAdapter {
+
+    public static final String SOAP_NAMESPACE =
+            "http://brooks/SCSS.Source.CeisScss.ws.provider:CeisScss";
```
Comment (HypnoMama) [inline src/main/java/ca/bc/gov/open/Scss/Configuration/SoapConfig.java:31]: @KFloodBCGov is this okay to be here?

File: src/main/java/ca/bc/gov/open/Scss/Configuration/SoapConfig.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+    @Bean
+    public RestTemplate restTemplate() {
+        RestTemplate restTemplate = new RestTemplate();
+        restTemplate.getMessageConverters().add(0, createMappingJacksonHttpMessageConverter());
+        return restTemplate;
+    }
+
+    private MappingJackson2HttpMessageConverter createMappingJacksonHttpMessageConverter() {
+
```
Comment (HypnoMama) [inline src/main/java/ca/bc/gov/open/Scss/Configuration/SoapConfig.java:50]:
```suggestion
```

File: src/main/java/ca/bc/gov/open/Scss/Configuration/SoapConfig.java [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    }
+
+    @Bean(name = "SCSS.Source.CeisScss.ws.provider:CeisScss")
+    public DefaultWsdl11Definition Wsdl11Definition() {
+        DefaultWsdl11Definition wsdl11Definition = new DefaultWsdl11Definition();
+        wsdl11Definition.setPortTypeName("scss");
+        wsdl11Definition.setCreateSoap11Binding(true);
+        wsdl11Definition.setLocationUri("/ws");
+        wsdl11Definition.setTargetNamespace(
+                "http://brooks/SCSS.Source.CeisScss.ws.provider:CeisScss");
```
Comment (HypnoMama) [inline src/main/java/ca/bc/gov/open/Scss/Configuration/SoapConfig.java:82]: convert to env

File: src/main/java/ca/bc/gov/open/Scss/Configuration/WSDLFilter.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+import javax.servlet.*;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+import org.springframework.stereotype.Component;
+
+@Component
+public class WSDLFilter implements Filter {
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
```
Comment (HypnoMama) [inline src/main/java/ca/bc/gov/open/Scss/Configuration/WSDLFilter.java:13]: Remove unused function

File: src/main/java/ca/bc/gov/open/Scss/Configuration/WSDLFilter.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+                            return super.getRequestURI() + ".wsdl";
+                        }
+                    };
+            chain.doFilter(requestWrapper, response);
+        } else {
+            chain.doFilter(request, response);
+        }
+    }
+
+    @Override
```
Comment (HypnoMama) [inline src/main/java/ca/bc/gov/open/Scss/Configuration/WSDLFilter.java:40]: remove unused function

File: src/main/java/ca/bc/gov/open/Scss/Controllers/CourtController.java [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+public class CourtController {
+
+    @Value("${scss.host}")
+    private String host = "http://127.0.0.1/";
+
+    private final RestTemplate restTemplate;
+
+    @Autowired
+    public CourtController(RestTemplate restTemplate) {
+
```
Comment (HypnoMama) [inline src/main/java/ca/bc/gov/open/Scss/Controllers/CourtController.java:27]:
```suggestion
```

File: src/main/java/ca/bc/gov/open/Scss/Controllers/CourtController.java [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+        this.restTemplate = restTemplate;
+    }
+
+    @PayloadRoot(namespace = SoapConfig.SOAP_NAMESPACE, localPart = "getCourtFile")
+    @ResponsePayload
+    public GetCourtFileResponse getCourtFile(@RequestPayload GetCourtFile search) {
+
```
Comment (HypnoMama) [inline src/main/java/ca/bc/gov/open/Scss/Controllers/CourtController.java:34]:
```suggestion
```

File: src/main/java/ca/bc/gov/open/Scss/Controllers/CourtController.java [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+                restTemplate.exchange(
+                        builder.toUriString(),
+                        HttpMethod.GET,
+                        new HttpEntity<>(new HttpHeaders()),
+                        GetPartiesResponse.class);
+
+        return resp.getBody();
+    }
+
+    @PayloadRoot(namespace = SoapConfig.SOAP_NAMESPACE, localPart = "partyNameSearch")
```
Comment (HypnoMama) [inline src/main/java/ca/bc/gov/open/Scss/Controllers/CourtController.java:102]: null pointer exception possible - please fix

File: src/main/java/ca/bc/gov/open/Scss/Controllers/CourtController.java [hunk 11]
```diff
@@ -0,0 +1,145 @@
...
+        pns.setSearchResults(resp.getBody());
+        return pns;
+    }
+
+    @PayloadRoot(namespace = SoapConfig.SOAP_NAMESPACE, localPart = "saveHearingResults")
+    @ResponsePayload
+    public SaveHearingResultsResponse saveHearingResults(
+            @RequestPayload SaveHearingResults search) {
+        UriComponentsBuilder builder =
+                UriComponentsBuilder.fromHttpUrl(host + "PartyNameSearch")
```
Comment (HypnoMama) [inline src/main/java/ca/bc/gov/open/Scss/Controllers/CourtController.java:133]: copy paste error?

File: src/main/java/ca/bc/gov/open/Scss/Controllers/FileController.java [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+        return new UnlinkFileResponse();
+    }
+
+    @PayloadRoot(namespace = SoapConfig.SOAP_NAMESPACE, localPart = "fileNumbeSearchPublicAccess")
+    @ResponsePayload
+    public FileNumbeSearchPublicAccessResponse fileNumberSearchPublicAccess(
+            @RequestPayload FileNumbeSearchPublicAccess search) {
+
+        UriComponentsBuilder builder =
+                UriComponentsBuilder.fromHttpUrl(host + "FileNumberSearch")
```
Comment (HypnoMama) [inline src/main/java/ca/bc/gov/open/Scss/Controllers/FileController.java:98]: add "PublicAccess"  and null pointer issue

File: src/main/java/ca/bc/gov/open/Scss/Models/Serializers/InstantSerializer.java [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+    public static Instant parse(String value) {
+        try {
+            Date d;
+            try {
+                d = new SimpleDateFormat("dd-MMM-yy hh.mm.ss.SSSSSS a", Locale.US).parse(value);
+            } catch (ParseException ex) {
+                d = new SimpleDateFormat("dd-MMM-yy", Locale.US).parse(value);
+            }
+            return d.toInstant();
+
```
Comment (HypnoMama) [inline src/main/java/ca/bc/gov/open/Scss/Models/Serializers/InstantSerializer.java:25]:
```suggestion
```

File: src/main/java/ca/bc/gov/open/Scss/ScssApplication.java [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+@Slf4j
+public class ScssApplication {
+
+    public static void main(String[] args) {
+
```
Comment (HypnoMama) [inline src/main/java/ca/bc/gov/open/Scss/ScssApplication.java:12]:
```suggestion
```

File: src/main/java/ca/bc/gov/open/Scss/ScssApplication.java [hunk 15]
```diff
@@ -0,0 +1,16 @@
...
+        SpringApplication.run(ScssApplication.class, args);
+        log.error("I am up");
```
Comment (HypnoMama) [inline src/main/java/ca/bc/gov/open/Scss/ScssApplication.java:14]:
```suggestion
```

File: src/main/resources/logback-test.xml [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+            <pattern>%-20(%d{HH:mm:ss.SSS} [%thread]) %-50(%-5level %logger:) %msg%n</pattern>
+        </encoder>
+
+        <!-- Show debug in stdout -->
+        <filter class="ch.qos.logback.classic.filter.ThresholdFilter">
+            <level>INFO</level>
+        </filter>
+    </appender>
+
+
```
Comment (HypnoMama) [inline src/main/resources/logback-test.xml:15]: Needs cleaning up

File: src/main/resources/xsdSchemas/RequestAndResponseObjects.xsd [hunk 17]
```diff
@@ -0,0 +1,402 @@
...
+
+    <xsd:element name="fileNumbeSearchPublicAccessResponse">
+        <xsd:complexType>
+            <xsd:sequence>
+                <xsd:element name="courtFiles" nillable="true" type="tns:CourtFile" maxOccurs="unbounded"/>
+            </xsd:sequence>
+        </xsd:complexType>
+    </xsd:element>
+
+</xsd:schema>
```
Comment (HypnoMama) [inline src/main/resources/xsdSchemas/RequestAndResponseObjects.xsd:402]: newline!!

File: src/test/java/ca/bc/gov/open/Scss/CourtControllerTests.java [hunk 18]
```diff
@@ -0,0 +1,237 @@
...
+
+    @Test
+    public void saveHearingResultTest() {
+        //  Init service under test
+        courtController = new CourtController(restTemplate);
+
+        ResponseEntity<String> responseEntity = new ResponseEntity<>("", HttpStatus.OK);
+
+        var hr = new SaveHearingResults();
+        //    var res = new HearingResult();
```
Comment (HypnoMama) [inline src/test/java/ca/bc/gov/open/Scss/CourtControllerTests.java:210]: If this is needed lets fill out and get it working.

File: src/test/java/ca/bc/gov/open/Scss/FileControllerTests.java [hunk 19]
```diff
@@ -0,0 +1,202 @@
...
+
+        //     Do request
+        var out = fileController.fileNumberSearchPublicAccess(fs);
+
+        //     Assert response is correct
+        assert (out.equals(fns));
+    }
+
+    @Test
+    public void securityTestFail_Then403() throws Exception {
```
Comment (HypnoMama) [inline src/test/java/ca/bc/gov/open/Scss/FileControllerTests.java:180]: remove test data

**Review by HypnoMama — COMMENTED** [review]
Nice work Ethan!

File: src/main/java/ca/bc/gov/open/Scss/Configuration/SoapConfig.java (hunk 3 above)
Comment (KFloodBCGov) [reply src/main/java/ca/bc/gov/open/Scss/Configuration/SoapConfig.java:31]: Yeah, this is fine.

File: src/main/java/ca/bc/gov/open/Scss/Configuration/SecurityConfig.java (hunk 2 above)
Comment (ekwanNTT) [reply src/main/java/ca/bc/gov/open/Scss/Configuration/SecurityConfig.java:18]: Fixed

File: src/main/java/ca/bc/gov/open/Scss/Configuration/SoapConfig.java (hunk 5 above)
Comment (ekwanNTT) [reply src/main/java/ca/bc/gov/open/Scss/Configuration/SoapConfig.java]: Fixed

File: src/main/java/ca/bc/gov/open/Scss/ScssApplication.java (hunk 15 above)
Comment (ekwanNTT) [reply src/main/java/ca/bc/gov/open/Scss/ScssApplication.java]: Fixed

File: src/main/resources/xsdSchemas/RequestAndResponseObjects.xsd (hunk 17 above)
Comment (ekwanNTT) [reply src/main/resources/xsdSchemas/RequestAndResponseObjects.xsd]: Fixed

File: src/test/java/ca/bc/gov/open/Scss/FileControllerTests.java (hunk 19 above)
Comment (ekwanNTT) [reply src/test/java/ca/bc/gov/open/Scss/FileControllerTests.java:180]: Fixed

File: src/test/java/ca/bc/gov/open/Scss/CourtControllerTests.java (hunk 18 above)
Comment (ekwanNTT) [reply src/test/java/ca/bc/gov/open/Scss/CourtControllerTests.java]: Filled out

**Review by HypnoMama — APPROVED** [review]

**Pull request merged**